src/database.rs

```diff
@@ -59,14 +60,19 @@ pub trait Database: Send + ZalsaDatabase + AsDynDatabase {
         zalsa_mut.runtime_mut().report_tracked_write(durability);
     }
 
-    /// This method triggers cancellation.
+    /// This method cancels all outstanding computations.
     /// If you invoke it while a snapshot exists, it
     /// will block until that snapshot is dropped -- if that snapshot
     /// is owned by the current thread, this could trigger deadlock.
     fn trigger_cancellation(&mut self) {
         let _ = self.zalsa_mut();
     }
 
+    /// Retrives a [`CancellationToken`] for the current database.
+    fn cancellation_token(&self) -> CancellationToken {
+        self.zalsa_local().cancellation_token()
+    }
```

Comment:
The `database` terminology is a bit misleading here. The way I think about the database is that there's only a single database, even if multiple owned instances refer to it after cloning it. 

While using the terminology thread is sort of inaccurate too because the database can move across threads, I think it would give me a better sense of what's going on.